### PR TITLE
refactor function update flow

### PR DESCRIPTION
**Pull request by edwardfoyle**
Refactor existing function update code to prepare for adding layers to functions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/execPermissionsWalkthrough.ts [hunk 1]
```diff
@@ -3,12 +3,18 @@ import inquirer from 'inquirer';
...
+import {
+  fetchPermissionCategories,
+  fetchPermissionResourcesForCategory,
+  fetchPermissionsForResourceInCategory,
+} from '../utils/permissionMapUtils';
+import { FunctionParameters, FunctionDependency } from 'amplify-function-plugin-interface/src';
 
 /**
  * This whole file desperately needs to be refactored
  */
```
Comment (jhockett) [inline packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/execPermissionsWalkthrough.ts:16]: Can we remove this now 🤣 ?

File: packages/amplify-category-function/src/provider-utils/awscloudformation/utils/permissionMapUtils.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  return _.keys(permissionMap);
+}
+
+export const fetchPermissionResourcesForCategory = (permissionMap, category: string) => {
+  return _.keys(_.get(permissionMap, [category]));
+}
+
+export const fetchPermissionsForResourceInCategory = (permissionMap, category: string, resourceName: string) => {
+  return _.get(permissionMap, [category, resourceName])
+}
```
Comment (jhockett) [inline packages/amplify-category-function/src/provider-utils/awscloudformation/utils/permissionMapUtils.ts:13]: nit: `;` after line 12, newline at end of file

File: packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/execPermissionsWalkthrough.ts (hunk 1 above)
Comment (edwardfoyle) [reply packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/execPermissionsWalkthrough.ts:16]: eh, I improved it, but it's by no means a finished product

File: packages/amplify-category-function/src/provider-utils/awscloudformation/utils/permissionMapUtils.ts (hunk 2 above)
Comment (edwardfoyle) [reply packages/amplify-category-function/src/provider-utils/awscloudformation/utils/permissionMapUtils.ts:13]: will fix that and push

**Pull request merged**